File: Jump Game.cpp
class Solution {
public:
    bool canJump(int A[], int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int last = 0;
        for (int i = 0; i < n; i++) {
            if (i > last) return false;
            last = max(last, i + A[i]);
        }
        return true;
    }
};

File: Jump Game.cs
public class Solution {
    public bool CanJump(int[] nums) {
        var last = 0;
        for (var i = 0; i < nums.Length; i++) {
            if (i > last) {
                return false;
            }
            last = Math.Max(last, i + nums[i]);
        }
        
        return true;
    }
}
